optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectEditor.java

```diff
@@ -121,7 +129,20 @@ protected void loadDataObject( DataObject dataObject ) {
         this.comparatorObject = null;
         if ( dataObject != null ) {
             boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
-            boolean hasPlanningSolution = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_SOLUTION_ANNOTATION ) != null;
+            boolean hasPlanningSolution = dataObject.getAnnotation( PLANNING_SOLUTION_ANNOTATION ) != null;
+
+            if ( !hasPlanningSolution ) {
+                dataModelerService.call( new RemoteCallback<List<Path>>() {
+                    @Override
+                    public void callback( List<Path> paths ) {
+                        view.enablePlanningSolutionCheckBox( paths.isEmpty() );
```

Comment:
@ge0ffrey The idea of using an artificial latency to expose n+1 query problems sounds interesting, CC'ing @jhrcek.

In this case, however, it wouldn't help because the current trend is to avoid using Selenium tests for checking validation, corner cases and negative testing (e.g. "given this and that, the checkbox should be disabled"). Due to maintenance cost we try to limit Selenium tests to happy path testing and end-to-end use cases.